### DISCUSSION TITLE
check_model for multiple models?

### DISCUSSION THREAD

**Discussion by candour23** [easystats/performance#293] [Q&A] [answered]
Hi,

This is my first brave question on github. First of all - great package- thanks a lot!

Second, I am running multiple linear regressions, but my knowledge is vey limited in programming and purr. 
I found a helpful site that helped me run multiple regression models (different outcomes) here: https://aosmith.rbind.io/2019/07/22/automate-model-fitting-with-loops/

But then, I wanted to use your beautiful package to check my assumptions on many models (different outcomes)... and I wasn't able to pull it off.
I would need to fit one by one model, and plot one by one (which is great until you have >3 models).

My ideal scenario would be having models in table as the link above explains (which I have),
and then using your fancy package to plot all these models in a single pdf, with names of output (for each separate model) corresponding to each of model, so I could easily detect something off.

I know this is going a bit beyond the scope, but if you have some suggestions, would be appreciated. 
Or if you know an easy way to integrate multiple models in check_model + name of each different outcome? 
Kind of just continuing on the first link I provided and relating it with check_model function?
Thanks in any case, great package!

**Comment by bwiernik** [comment, marked as answer]
So once you've got your list of model objects `model_list` --`lapply(model_list, check_model)` or `map(model_list, check_model)` will apply the `check_model` function to each model in the list.

**Reply by candour23** [reply]
Thanks a lot for your promptly reply, but if I am not mistaken, I will still not be able to see which plot corresponds to which model?
To somehow also extract the names of the models within this?

**Reply by bwiernik** [reply]
assign that result as an object `plot_list <- map(...)` and then extract the results by name `plot_list$mpg`

**Reply by candour23** [reply]
I am not sure if I am not grasping your concept, or if you are on the same page with me for the question.
For each model I will have 5 plots (just like first solution you gave), what would be useful is to above these (or anywhere) an actual name of the plot (Model 1) above the plot itself, then for next 5 plots- Model 2, etc.
Can this be embedded on the overall plot itself? 
So, if I am checking 25 models from single pdf or however I save it, and I see one really odd, from the pdf itself I can say this exact model with this outcome (written above it) is violating my assumptions?

**Reply by bwiernik** [reply]
At the moment, I suggest using headings in your RMarkdown document to do this, ala:

![image](https://user-images.githubusercontent.com/4773225/117833093-0d901a00-b244-11eb-971c-e6c0ff584f09.png)

```{r}
library(purrr)
library(patchwork)
library(performance)

model_list <- map(
  c(disp = "disp", hp = "hp", drat = "drat", wt = "wt"), 
  ~ lm(as.formula(paste("mpg ~ ", .x)), data = mtcars)
)

plot_list <- map(
  model_list,
  check_model
)
```

## Model Checking

### disp

```{r}
plot_list$disp
```

### hp

```{r}
plot_list$hp
```

**Reply by bwiernik** [reply]
Doing more customization will be easier if *see* switches to using the *patchwork* package instead https://github.com/easystats/performance/discussions/293#discussioncomment-723980

**Reply by candour23** [reply]
Okay, if markdown cant be avoided, then I'll implement it as you suggested.
Thanks a lot for looking into it and giving me a temporary solution.
